### ISSUE TITLE
"SELECT cid,rid FROM seen WHERE uid=1 AND ts>0  LIMIT 3;" ERROR:  Failed to execute query: SELECT cid,rid FROM seen WHERE uid=1 AND ts>0  LIMIT 3;, source: Failed to execute logical plan, source: Failure during query execution, source: Fail to crea te physical plan: type_coercion

### ISSUE THREAD

**Issue by xxaier** [GreptimeTeam/greptimedb#2026]
### What type of bug is this?

Unexpected error

### What subsystems are affected?

Standalone mode

### What happened?

```sql
public=> SELECT cid,rid FROM seen WHERE uid=1 LIMIT 3;
 cid | rid
-----+------
 2   | 7822
 2   | 8079
 2   | 9575
(3 行记录)

public=> SELECT cid,rid FROM seen WHERE uid=1 AND ts>0  LIMIT 3;
ERROR:  Failed to execute query: SELECT cid,rid FROM seen WHERE uid=1 AND ts>0  LIMIT 3;, source: Failed to execute logical plan, source: Failure during query execution, source: Fail to crea
te physical plan: type_coercion
caused by
Error during planning: Timestamp(Millisecond, None) > Int64 can't be evaluated because there isn't a common type to coerce the types to
public=> SELECT cid,rid FROM seen WHERE uid=1 AND uid>0  LIMIT 3;
 cid | rid
-----+------
 2   | 7822
 2   | 8079
 2   | 9575
(3 行记录)

public=> SELECT cid,rid FROM seen WHERE uid=1 AND ts>0  LIMIT 3;
ERROR:  Failed to execute query: SELECT cid,rid FROM seen WHERE uid=1 AND ts>0  LIMIT 3;, source: Failed to execute logical plan, source: Failure during query execution, source: Fail to crea
te physical plan: type_coercion
caused by
Error during planning: Timestamp(Millisecond, None) > Int64 can't be evaluated because there isn't a common type to coerce the types to
```

Slack chat history:

[11:40](https://greptimedbcommunity.slack.com/archives/C03TY50Q682/p1690256443138249)
我又发现一个bug，ts>数值 和 limit 不能一起用
[11:41](https://greptimedbcommunity.slack.com/archives/C03TY50Q682/p1690256493849849)
SELECT cid,rid FROM seen WHERE uid=1 AND ts>CAST(0 as timestamp)  LIMIT 3; 要这样写才行
[11:42](https://greptimedbcommunity.slack.com/archives/C03TY50Q682/p1690256535695499)
不加limit的时候可以写 SELECT cid,rid FROM seen WHERE uid=1 AND ts>0;

### What operating system did you use?

any

### How can we reproduce the bug?

SELECT cid,rid FROM seen WHERE uid=1 AND ts>0  LIMIT 3;

**Comment by xxaier** [comment]
CAST转换貌似也不对，这个查询应该感觉不应该包含 1690252336408
English (USA): CAST's conversion seems wrong, this query should not be included 169025236408

![image](https://github.com/GreptimeTeam/greptimedb/assets/130831741/22b320ae-39a0-45ac-b697-04bd101a2fef)

**Comment by waynexia** [comment]
Submitted a doc PR to include this case https://github.com/GreptimeTeam/docs/pull/500. Thanks for reporting
